pkg/client/options.go

```diff
@@ -40,10 +42,15 @@ func getClientOpts(opts ...psrpc.ClientOption) psrpc.ClientOpts {
 	return *o
 }
 
-func getRequestOpts(i *info.RequestInfo, options psrpc.ClientOpts, opts ...psrpc.RequestOption) psrpc.RequestOpts {
+func getRequestOpts(ctx context.Context, i *info.RequestInfo, options psrpc.ClientOpts, opts ...psrpc.RequestOption) psrpc.RequestOpts {
 	o := &psrpc.RequestOpts{
 		Timeout: options.Timeout,
 	}
+	if o.Timeout == psrpc.DefaultClientTimeout {
```

Comment:
The way I understand typical context usage, I would expect the resulting timeout to always be the shorter of the context and psrpc timeout.

Independently, would it make sense to honor the context getting canceled regardless of the cancellation reason? (timeout, explicit cancellation, etc)

Reply:
I considered this case too, but the main reason I needed this change was to increase the timeout, not reduce it. 

And this is exactly the reason why honoring context cancellation doesn't work - it cancels the request _before_ the parent context is cancelled.

Reply:
This seems like an unexpected use of the context semantics that may become confusing for folks using psrpc. What is the use case here? Why can't we increase/disable the psrpc timeout provided in the options and rely only on the context?

Reply:
Indeed, I would prefer to disable the implicit timeout in psrpc. So if not timeout is set via option and the context, the client default timeout would apply. If option or context timeout are set - they are used. And if both are set - the shortest one applies. I believe this is how gRPC does it.

I was just a bit reluctant to set default value of `ClientOptions.Timeout` to `0`. Not sure, maybe it can break some existing code? And without this, the only way to check for "timeout not set explicitly" is to check for a default value as I did in the PR. So yes, if it is okay to remove the default from `ClientOptions`, I'd rather do that.

Reply:
Is there any reason not to always apply the default timeout, even if it was not explicitly set, and it is shorter than the context timeout?

Reply:
Yes, as I said, my original issue was that I was sending a request that in normal circumstances took longer than the default timeout for psrpc. So the only way to make it work was to set timeout explicitly. At first I did it via context, which didn't work because psrpc doesn't honor it. I'd really prefer to use it the same way as I'd use other libraries like gRPC, so that the context timeout is used even if it's longer than the default one.